### PR TITLE
"HTTP testing in R" book update

### PR DESCRIPTION
**Pull request by maelle**
@sckott would you be able to have a look at this? Not urgent, no publication date is set.

### CODE REVIEW

**Comment by maelle** [comment]
* [ ] I have read the Content Guidelines.
* [ ] I have read the Technical Guidelines.
* [ ] I used or followed the R Markdown or Markdown template.
* [ ] I have followed the Style Guide.
* [ ] I created or updated my author metadata with correct folder name.
* [ ] I have added relevant tags after browsing existing tags.
* [ ] I ran `roblog::ro_lint_md()` on index.md (optional).
* [ ] I ran `roblog::ro_check_urls()` on index.md (optional).
* [ ] I ran a spell-check on index.md.

**Comment by maelle** [comment]
I am wondering whether including one or a few positive reactions from Twitter would be nice or look bad.

**Comment by maelle** [comment]
Maybe I should add links to the PDF and EPUB version as they are only on the first page at the moment.

File: content/blog/2021-02-02-http-testing-book/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+* A [conclusion](https://books.ropensci.org/http-testing/conclusion-4.html) (again, how original) underlining what the next steps are. In particular, we encourage you to read the docs of the HTTP package(s) you end up using. For [webmockr](https://books.ropensci.org/http-testing/mocking.html) and [vcr](https://books.ropensci.org/http-testing/vcr-intro.html), these docs also live in the same book so you don't have far to go.
+
+
+## Other technical details
+
+Whilst writing chapters Maëlle filled several docs and feature requests to vcr, httptest, webfakes.
+All three maintainers were very cooperative and nice.
+
+* There was a recent tech note about vcr release, part of which is a consequence of the book update.
```
Comment (sckott) [inline content/blog/2021-02-02-http-testing-book/index.Rmd:72]:
```suggestion
* There was a recent [tech note about vcr release](https://ropensci.org/technotes/2020/12/18/vcr-release/), part of which is a consequence of the book update.
```

**Review by sckott — COMMENTED** [review]
looking good! just one suggestion

**Comment by maelle** [comment]
Thanks @sckott!

**Comment by stefaniebutland** [comment]
Can be published 2021-01-19. Maëlle tells me the post will be ready after @steffilazerte's review.

**Comment by steffilazerte** [comment]
Great! I'll take a look on Monday.

**Comment by steffilazerte** [comment]
So I did not look at this on Monday (eep), and as the publication date isn't until 2021-01-26 I hope it's okay that I don't look at this until this Monday/Tuesday 😅  Sorry!

File: content/blog/2021-02-02-http-testing-book/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Scott created the book a while back, as central documentation for his packages [crul](https://docs.ropensci.org/crul), [vcr](https://docs.ropensci.org/) and [webmockr](https://docs.ropensci.org/).
+Maëlle got [funding from the R Consortium](https://www.r-consortium.org/projects/awarded-projects/2020-group-1#HTTP+testing+in+R+Book) to expand the scope of the book.
+Thank you, R Consortium!
+Thanks also to Scott, [Adam Sparks](/author/adam-sparks/), [Matthias Grenié](/author/matthias-grenié/) and [Neal Richardson](https://enpiar.com/) for contributing to the project proposal. Thanks also to [Gábor Csárdi](https://github.com/gaborcsardi) who was also consulted... even before he created webfakes!
+Last but not least, thanks to recent readers of the book like [Dave Parr](https://www.daveparr.info/), [Lluís Revilla Sancho](https://llrs.dev/) and [Hugo Gruson](https://www.normalesup.org/~hgruson/) for useful feedback.
+
+## What's between the covers?
+
+The book still has advanced content about webmockr and vcr, but these only come after new content.
```
Comment (steffilazerte) [inline content/blog/2021-02-02-http-testing-book/index.Rmd:46]:
```suggestion
The book still has advanced content about webmockr and vcr, but now this comes after new content.
```

File: content/blog/2021-02-02-http-testing-book/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+* An introduction (how original), giving some basic information and resources suggestions around [HTTP in R](https://books.ropensci.org/http-testing/http-in-r-101.html) and [_graceful HTTP R packages_](https://books.ropensci.org/http-testing/graceful.html) as well as the reason why we need [special packages for HTTP testing](https://books.ropensci.org/http-testing/pkgs-testing-chapter.html).
```
Comment (steffilazerte) [inline content/blog/2021-02-02-http-testing-book/index.Rmd:48]: Should the second link be italicized?

```suggestion
* An introduction (how original), giving some basic information and resource suggestions around [HTTP in R](https://books.ropensci.org/http-testing/http-in-r-101.html) and [graceful HTTP R packages](https://books.ropensci.org/http-testing/graceful.html) as well as the reason why we need [special packages for HTTP testing](https://books.ropensci.org/http-testing/pkgs-testing-chapter.html).
```

File: content/blog/2021-02-02-http-testing-book/index.Rmd [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+   * [Security](https://books.ropensci.org/http-testing/security-chapter.html);
+   * [Faking HTTP errors](https://books.ropensci.org/http-testing/errors-chapter.html);
+   * [Contributor friendliness](https://books.ropensci.org/http-testing/contributor-friendliness.html).
+
+* A [conclusion](https://books.ropensci.org/http-testing/conclusion-4.html) (again, how original) underlining what the next steps are. In particular, we encourage you to read the docs of the HTTP package(s) you end up using. For [webmockr](https://books.ropensci.org/http-testing/mocking.html) and [vcr](https://books.ropensci.org/http-testing/vcr-intro.html), these docs also live in the same book so you don't have far to go.
+
+
+## Other technical details
+
+Whilst writing chapters Maëlle filled several docs and feature requests to vcr, httptest, webfakes.
```
Comment (steffilazerte) [inline content/blog/2021-02-02-http-testing-book/index.Rmd:69]:
```suggestion
Whilst writing chapters Maëlle filed several docs and feature requests to vcr, httptest, webfakes.
```

File: content/blog/2021-02-02-http-testing-book/index.Rmd [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+* In httptest the new `httptest::with_mock_dir()` function was contributed as a result of Maëlle's work.
+* In webfakes Maëlle suggested and helped develop the [OAuth2.0 apps](http://webfakes.r-lib.org/articles/oauth.html).
+
+Besides, the book itself is an early adopter of a brand-new bookdown template by Hadley Wickham, `bookdown::bs4_book()`, which you might want to check out for your own bookish endeavors!
+It's only available in the [development version of bookdown](https://github.com/rstudio/bookdown/) at the moment of writing, and there quite a [few features still in the work](https://github.com/rstudio/bookdown/issues?q=is%3Aopen+sort%3Aupdated-desc+label%3A%22bs4_book+%3Ahiking_boot%3A%22+).
+As for deployment, we'd recommend looking at the [GitHub Actions workflow for the R packages book](https://github.com/hadley/r-pkgs/blob/master/.github/workflows/build-book.yaml) that we drew inspiration from.
+
+## Further work and feedback request!
+
+While we are happy of the book current state, we are committed to updating and improving it!
```
Comment (steffilazerte) [inline content/blog/2021-02-02-http-testing-book/index.Rmd:82]:
```suggestion
While we are happy with our "HTTP testing in R" book, we are committed to updating and improving it!
```

File: content/blog/2021-02-02-http-testing-book/index.Rmd [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+   * [Faking HTTP errors](https://books.ropensci.org/http-testing/errors-chapter.html);
+   * [Contributor friendliness](https://books.ropensci.org/http-testing/contributor-friendliness.html).
+
+* A [conclusion](https://books.ropensci.org/http-testing/conclusion-4.html) (again, how original) underlining what the next steps are. In particular, we encourage you to read the docs of the HTTP package(s) you end up using. For [webmockr](https://books.ropensci.org/http-testing/mocking.html) and [vcr](https://books.ropensci.org/http-testing/vcr-intro.html), these docs also live in the same book so you don't have far to go.
+
+
+## Other technical details
+
+Whilst writing chapters Maëlle filled several docs and feature requests to vcr, httptest, webfakes.
+All three maintainers were very cooperative and nice.
```
Comment (steffilazerte) [inline content/blog/2021-02-02-http-testing-book/index.Rmd:70]: For some reason, "cooperative" has a slightly formal tone, I think receptive might be a nicer way of phrasing it. 

```suggestion
All three maintainers were very kind and receptive.
```

**Review by steffilazerte — COMMENTED** [review]
Hi @maelle!
Thanks for your patience, the last couple of weeks have been busy! I enjoyed reading the blog post and made numerous mental notes to check it out for my own testing on weathercan :grin:

Everything looks good! I have a couple of wording suggestions and the publication date needs to be updated, but other than that everything looks good. 

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [X] title is in Title Case
* [ ] publication date is ok
   - **2021-01-26 (YAML and folder)**
* [X] alternative text of images is informative
* [X] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
* [X] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok (add "community" for post by non-staff non-editor)

**Comment by steffilazerte** [comment]
Tweet draft, comments welcome!

---

[blog] Learn All About HTTP Testing: Book Update 📘

🎉 @ma_salmon gives an overview of updates such as

- Setup examples with different HTTP testing packages 
- 📦 Package comparisons
- Advanced topics

https://ropensci.org/blog/2021/01/26/http-testing-book/

#rstats #vcr #webmockr #httptest #webfakes #testing

**Comment by steffilazerte** [comment]
Looks good for publishing tomorrow! And good call on the tweets, I'm never that creative so I really appreciate your ideas! If you want to do the tweet(s) yourself, just let me know and go ahead!

Otherwise, what do you think of the following (I'll tweet around 10:00 Central, 17:00 France)

---
Do you
... develop an #rstats package interfacing web resources?
... wish you could improve its unit tests with state-of-the-art HTTP testing tools?

The HTTP testing in R book by @sckottie and @ma_salmon could help!

Read about the book new content in:

https://ropensci.org/blog/2021/01/26/http-testing-book

(1/3)

---
[blog] Learn All About HTTP Testing: Book Update 📘

🎉 @ma_salmon and @sckottie present an overview of updates

- Setup examples with different HTTP testing packages 
- 📦 Package comparisons
- Advanced topics

https://ropensci.org/blog/2021/01/26/http-testing-book/

#rstats #vcr #webmockr #httptest #webfakes #testing

(2/3)

---
🙏 Thank yous go out to 

- @RConsortium for funding
- @adamhsparks, @LeNematode, @enpiar and @GaborCsardi for contributing to the project proposal
- @DaveParr @Lluis_Revilla and @grusonh for feedback on the book

(3/3)

**Pull request merged**